### DISCUSSION TITLE
Send FIL to my Filecoin ID address using MetaMask

### DISCUSSION THREAD

**Discussion by longfeiWan9** [filecoin-project/community#649] [Q&A] [answered]
If you are devs or users who are familiar with Filecoin address types, you probably already know that Filecoin supports [multiple address types](https://docs.filecoin.io/smart-contracts/filecoin-evm-runtime/address-types/).
In short, Filecoin addresses type include:
- **ID address**: prefix is `f0/t0`. short ID created for any address on Filecoin network.  e.g. `t01001`
- **Regular wallet address**: prefix is `f1/t1` and `f3/t3`. e.g. `t1ssd2ydngi3tcipvcwx4wdyaseqgwbt2ivjis4qi`
- **Actor Address**: prefix is `f2/t2` represents actor deployed on Fielcoin. e.g. `t2kmbjvz7vagl2z6pfrbjoggrkjofxspp7cqtw2zy`
- **Delegate address** for foreign actors
    - **EVM** foreign address: prefix is `f410/t410`. e.g. `t410f2oekwcmo2pueydmaq53eic2i62crtbeyuzx2gmy`

> Keep in mind that any address in the Filecoin network has a unique ID address assigned to it. 
This ID address will be finalized after 900 epochs. Potentially they still may be changed for chain reorg which happens rarely after 900 epochs. But be careful when using the ID address to send FIL. Never use a freshly generated ID address. 

## Having difficulty sending FIL to your f1 or f3 address using MetaMask?
Since f1 or f3 address can not be converted to the Ethereum-compatible 0x address, MeteMask will not be able to recognize and use f1 or f3 address. **BUT**, 
- ID address can be converted to 0x address
- all f1 and f3 addresses will have a unique ID address assigned to them. This ID address will be finalized after 900 epochs of creation.
<img width="623" alt="image" src="https://github.com/filecoin-project/community/assets/34025001/73d690df-5d8f-4dba-b33f-a4d251ba7257">

which means technically, you can send FIL to your f1 or f3 address using their corresponding ID address. For example, I can send FIL to `t03763` using MetaMask for the `t1qzpdbtq7wxkkghhaxtrffiimbws2ba7wp3zepea`. 

#### 1. Find the ID address for your f1 or f3 address. 
> Keep in mind, do not use ID addresses for freshly created f1 or f3 addresses. Wait until it is finalized after 900 epochs.
- you can find your ID address on any blockchain explorer, Filfox, Beryx, or Starboard FVM explorer. 
- you can find your ID address via Lotus Cli, `lotus wallet list -i`.
#### 2. Convert ID address to 0x address.
Once you have your ID address for your f1 or f3 address, you need to convert it to 0x address which you can use on MetaMask. The basic instruction for converting ID address to 0x address is [here](https://docs.filecoin.io/smart-contracts/filecoin-evm-runtime/address-types/#converting-to-a-0x-style-address). 
- you can also convert it using Lotus Cli.
  ```
  longfeiwang@Filecoin ~ $ lotus evm stat f03763
  
  Filecoin address:  f03763
  Eth address:  0xff00000000000000000000000000000000000eb3
  ID address:  f03763
  Code cid:  bafk2bzacealnlr7st6lkwoh6wxpf2hnrlex5sknaopgmkr2tuhg7vmbfy45so
  Actor Type:  fil/11/account
  ```
- hopefully, this conversion will be provided on Filecoin blockchain explorers so users can directly copy and use it. 
Then you will get `0xff00000000000000000000000000000000000eb3` which represents your Filecoin ID address `f03763`. 
#### 3. sending FIL using 
Now, you can directly use `0xff00000000000000000000000000000000000eb3` as to-address to transfer FIL using MetaMask. FIL will be transferred to your f1 or f3 wallet. 
<img width="371" alt="image" src="https://github.com/filecoin-project/community/assets/34025001/b3b9b854-b758-430e-9e0e-9ab27c415c2d">

Therefore, technically users or devs can send FIL to any Filecoin address type using its corresponding ID address using MetaMask.

**Comment by gorbunova13** [comment]
thanks

**Comment by longfeiWan9** [comment, marked as answer]
**One more note:**
 
Sending FIL to MinerID on MetaMask is not supported yet because the built-in miner actor doesn't understand `invokeEVM` method. And sending FIL to 0x address is `invokeEVM` method.
